src/spyglass/lfp/analysis/v1/lfp_band.py

```diff
@@ -109,57 +137,133 @@ def set_lfp_band_electrodes(
             lfp_band_sampling_rate=lfp_sampling_rate // decimation,
         )
-        # insert an entry into the main LFPBandSelectionTable
-        self.insert1(key, skip_duplicates=True)
-
-        key["lfp_electrode_group_name"] = lfp_part_table.fetch1(
+        # nwb_file_name, lfp_electrode_group_name, electrode_group_name, electrode_id, reference_elect_id
+        part_keys = []
+        lfp_electrode_group_name = lfp_part_table.fetch1(
             "lfp_electrode_group_name"
         )
-        # iterate through all of the new elements and add them
-        for e, r in zip(electrode_list, ref_list):
-            elect_key = (
-                LFPElectrodeGroup.LFPElectrode
-                & {
+
+        # get the electrode group names for the electrodes
+        electrode_group_name_list, check_electrode_id_list = (
+            LFPElectrodeGroup.LFPElectrode()
+            & [
+                {
                     "nwb_file_name": nwb_file_name,
-                    "lfp_electrode_group_name": key["lfp_electrode_group_name"],
-                    "electrode_id": e,
+                    "lfp_electrode_group_name": lfp_electrode_group_name,
+                    "electrode_id": electrode_id,
                 }
-            ).fetch1("KEY")
-            for item in elect_key:
-                key[item] = elect_key[item]
-            query = Electrode & {
-                "nwb_file_name": nwb_file_name,
-                "electrode_id": e,
-            }
-            key["reference_elect_id"] = r
-            self.LFPBandElectrode().insert1(key, skip_duplicates=True)
+                for electrode_id in electrode_list
+            ]
+        ).fetch("electrode_group_name", "electrode_id", order_by="electrode_id")
+
+        # check that the electrode ids match in order to avoid
+        # inserting the wrong electrode group names
+        if not np.array_equal(electrode_list, check_electrode_id_list):
+            raise ValueError(
+                "Electrode IDs do not match the electrode IDs in the "
+                + "LFPElectrodeGroup table"
+            )
+
+        for electrode_id, reference_elect_id, electrode_group_name in zip(
+            electrode_list, ref_list, electrode_group_name_list
+        ):
+            part_keys.append(
+                {
+                    **master_key,
+                    "lfp_electrode_group_name": lfp_electrode_group_name,
+                    "electrode_group_name": electrode_group_name,
+                    "electrode_id": electrode_id,
+                    "reference_elect_id": reference_elect_id,
+                }
+            )
+
+        # check if the LFPBandSelection table already has this entry
+        if self & master_key:
+            # if it does, check if the electrodes are the same
+            existing_electrodes = (self.LFPBandElectrode() & master_key).fetch(
+                "electrode_id"
+            )
+            if set(existing_electrodes) == set(electrode_list):
```

Comment:
I put together a `lfp.lfp_electrode.LFPElectrodeGroup.cautious_insert` for #1278 that covers the same logic for checking a master key, and then checking matching set. With that merged, you could reuse the logic here

Reply:
I think I'm going to leave this for later since I don't have time to implement.